cli/src/main.rs

```diff
@@ -262,35 +266,46 @@ fn ping() -> Result<()> {
     Ok(())
 }
 
-fn blocking_maybe_discover_service_url(
-    service_type: &str,
-    env_var_override_name: &str,
-) -> Result<String> {
-    if let Ok(override_url) = std::env::var(env_var_override_name) {
+async fn maybe_find_peer(role: &ServalRole, override_var: &str) -> Result<String> {
```

Comment:
Thoughts on initializing the mesh elsewhere and passing a reference to it into this function? (Or does the CLI only ever need to find a peer a single time?)